frontend/src/scenes/debug/QueryTabs.tsx

```diff
@@ -33,13 +33,13 @@ function toLine(hogql: string, position: number): number {
 function toColumn(hogql: string, position: number): number {
     return toLineColumn(hogql, position).column
 }
-interface QueryTabsProps {
-    query: Node
+interface QueryTabsProps<Q extends Node> {
+    query: Q
     queryKey: `new-${string}`
-    response?: Record<string, any> | null
-    setQuery: (query: DataNode) => void
+    response?: Q['response'] | null
```

Comment:
There's a bunch of changes like this. Thanks to better guarantees, specifically on `response` here, TypeScript was complaining before the switch to generics here. Generics are definitely the accurate way here though.